mlibrary/search
# ChooseAffiliation form method

# Overview
`ChooseAffiliation` has been problematic ever since [switching to the `Dialog` component](https://github.com/mlibrary/search/pull/388) for accessibility. This pull request cleans up the code to reduce repetition.

## Anything else?
The option to click outside of the `dialog` in order to close it has been removed. This is to help those who may have hand tremors or other difficulties using a mouse and may accidentally click outside of it.

### `react-cookie`
`react-cookie` has been replaced with native `localStorage` for two reasons: to rely on one less dependency and because the site does not contain a banner that informs the user that the site uses cookies.

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Load the site and click on `ANN ARBOR` or `FLINT`.
  - Does the `dialog` appear?
  - Does it switch to the other affiliation when you choose to switch?
  - Can you close the `dialog` by clicking `Dismiss`, pressing `Escape`, or continuing to use the same affiliation?
  - If you remove the `affiliation` query from the URL and load that URL, does the affiliation remain the same?
